### PR TITLE
MINOR: remove partition-level error from MetadataResponse#errorCounts

### PR DESCRIPTION
**Pull request by chia7712**
see https://github.com/apache/kafka/pull/9433#discussion_r676083224

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by ijuma** [comment]
Thanks for the PR. Are there other similar cases in the original PR or is this the only one?

**Comment by chia7712** [comment]
> Are there other similar cases in the original PR or is this the only one?

`StopReplicResponse` (https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/requests/StopReplicaResponse.java#L55) has similar issue. will take a look at others.

For another, I'm thinking about better code style. As server-side does not add partition error to response when top-level gets error, `errorCounts` should be fine to collect all error code from all elements. It seems to me the only challenge is that the count of `NONE` may get higher.

Could you share the regression you mentioned to me if I overlook the real problem?

**Comment by ijuma** [comment]
There are two problems:
1. The count for NONE would be wrong
2. Negative performance impact

**Comment by chia7712** [comment]
> Negative performance impact

this is a good reason.

> The count for NONE would be wrong

As the "scoped error" can be various between responses, we need to count NONE case by case. 

thanks for response. will update code later.

File: clients/src/main/java/org/apache/kafka/common/requests/MetadataResponse.java [hunk 1]
```diff
@@ -100,10 +100,7 @@ public int throttleTimeMs() {
     @Override
     public Map<Errors, Integer> errorCounts() {
         Map<Errors, Integer> errorCounts = new HashMap<>();
-        data.topics().forEach(metadata -> {
-            metadata.partitions().forEach(p -> updateErrorCounts(errorCounts, Errors.forCode(p.errorCode())));
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/common/requests/MetadataResponse.java:104]: According to https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/metadata/ZkMetadataCache.scala#L103, it is possible that partition-level has error but topic-level gets none. Hence, it produces inaccurate count of "error" if we remove this line.

If we want to make correct count of `error` and `none`, we have to add following check.

1. if topic has error, we don't need to loop partitions.
2. if topic has no error, we have to loop partitions.

@ijuma WDYT?

**Pull request closed without merging**